### ISSUE TITLE
ACLs for project admin and delete_execution are mutually exclusive

### ISSUE THREAD

**Issue by shvytejimas** [rundeck/rundeck#3012]
Issue type: Bug report

**My Rundeck detail**

* Rundeck version: v2.10.1
* install type: rpm
* OS Name/version: centos-7.3-x64

**Expected Behavior**

Have a working ACL such that users are able to modify project settings (add new inventory sources, for example), but at the same time unable to delete executions in the project (so that no activity is lost for auditing). 

**Actual Behavior**

The `configure` ACL permission has no effect on a project unless `admin` permission is also set. However enabling `admin` permission also overrides the `deny` clause on `delete_execution` permission. It is therefore not possible to separate project modification and execution deletion in terms of permissions.

**How to Reproduce**

On a fresh Install of v2.10.1 with a test user: `testuser:testpassword,user,admin` created a `testproject` and ran a few executions. Using the following rule:

```
description: Deny deleting executions
context:
  application: 'rundeck'
for:
  project:
    - deny: [ delete_execution ]
      equals:
        name: testproject
by:
  username: testuser
```

The button `Delete This Execution` is still visible and working in the UI. The audit log shows:

```
Evaluating Decision for: res<name:testproject, type:project> subject<Username:testuser Group:admin Group:user Group:testuser> action<delete_execution> env<rundeck:auth:env:application:rundeck>: authorized: false: REJECTED, reason: REJECTED_DENIED, evaluations:  ACLRule</etc/rundeck/testdeny.aclpolicy[1][type:project][rule: 1]>{'Deny deleting executions' context={application='rundeck'} type='project' equals , resource={name=testproject} for: { username='testuser'} deny=[delete_execution]} REJECTED_DENIED for action delete_execution => REJECTED_DENIED (0ms)
Evaluating Decision for: res<name:testproject, type:project> subject<Username:testuser Group:admin Group:user Group:testuser> action<admin> env<rundeck:auth:env:application:rundeck>: authorized: true: GRANTED, reason: GRANTED, evaluations:   ACLRule</etc/rundeck/admin.aclpolicy[2][type:project][rule: 1]>{'Admin, all access.' context={application='rundeck'} type='project' for: { group='admin'} allow=[*]} GRANTED for action admin => GRANTED (0ms)
```

Seems that `admin` permission overrides the `deny: delete_execution` clause. After modifying the ACL rule to remove `admin` permission for the user as such:

```
  project:
    - deny: [ admin, delete_execution ]
      equals:
        name: testproject
```

The `Delete This Execution` button is no longer visible. 

However, `Project > Edit Configuration` or `Project > Edit Nodes` menu drop-downs now result in an error ( `Not authorized to configure Project testproject` ). Adding the following to ACL rule:

```
project:
    - allow: configure
```

Has no effect since it seems (from the audit log) that `admin` permission is checked regardless of `configure` permission.

```
Evaluating Decision for: res<type:project, name:testproject> subject<Username:testuser Group:admin Group:user Group:testuser> action<configure> env<rundeck:auth:env:application:rundeck>: authorized: true: GRANTED, reason: GRANTED, evaluations:   ACLRule</etc/rundeck/admin.aclpolicy[2][type:project][rule: 1]>{'Admin, all access.' context={application='rundeck'} type='project' for: { group='admin'} allow=[*]} GRANTED for action configure => GRANTED (0ms)
Evaluating Decision for: res<type:project, name:testproject> subject<Username:testuser Group:admin Group:user Group:testuser> action<admin> env<rundeck:auth:env:application:rundeck>: authorized: false: REJECTED, reason: REJECTED_DENIED, evaluations:   ACLRule</etc/rundeck/testdeny.aclpolicy[1][type:project][rule: 1]>{'Deny deleting executions' context={application='rundeck'} type='project' equals , resource={name=testproject} for: { username='testuser'} deny=[delete_execution, admin]} REJECTED_DENIED for action admin => REJECTED_DENIED (0ms)
```

**Comment by cosmih** [comment]
Hi,

I am able to reproduce the issue with 2.10.2 
My setup:
- version 2.10.2
- install type: rpm
- OS Name/version: centos-7.4-x64 
- JVM vendor/version: OpenJDK 1.8.0.151

**Comment by gschueler** [comment]
Note: the current ACL checks require `admin` for editing a project config/node sources via the GUI, although only `configure` to do so via API.

The fix for this issue is to correctly allow `configure` to edit project config/node sources, without requiring `admin` level.

However, making this change could inadvertently increase access level for some users depending on the ACLs, if `admin` is expected as the required access level.

I will schedule this fix for 2.11.0, to avoid altering the existing access level settings for users of 2.10.
